### ISSUE TITLE
Numpy depreciated ComplexWarning

### ISSUE THREAD

**Issue by victoriyaforsythe** [SuperDARN/pydarn#384]
# BUG 

Hello! I was not able to import pydarn today. The attribute error states that numpy has no attribute 'ComplexWarning'.

## Priority

- [ ] Minor - typo, incorrect naming, fails on specific uncommon situation
- [ ] Major - misrepresentation of data, failure in a common situation 
- [V ] Urgent - failure in usage, large misrepresentation

## Information
**python version 3.12.3**:
**Mac Sonoma 14.5**:

<img width="617" alt="Screenshot 2024-06-24 at 11 15 34 AM" src="https://github.com/SuperDARN/pydarn/assets/116663397/a5048636-0bc3-4111-937d-5b223c678547">

**Comment by RemingtonRohel** [comment]
Hello! Do you have the steps that you used to install pydarn, and the versions of software (specifically pydarnio and numpy) that are installed?

**Comment by victoriyaforsythe** [comment]
I re-installed pydarn using pip:
pip3 install pydarn --break-system-packages --force-reinstall
I can't see what version it is, because I cannon import it, but I assume it is the latest one on PyPi. 

My Numpy version is 2.0.0

**Comment by RemingtonRohel** [comment]
Hmm. Could you send a list of all the packages you have installed, using the command `pip list`? I suspect it is coming from a dependency mismatch of some sort, not pydarn or pydarnio themselves.

**Comment by victoriyaforsythe** [comment]
It is quite a list:
Package            Version
------------------ -----------
aacgmv2            2.6.3
apexpy             2.0.1
appnope            0.1.4
asteval            0.9.32
asttokens          2.4.1
blosc2             2.7.0
build              1.2.1
Cartopy            0.23.0
certifi            2024.6.2
cftime             1.6.4
charset-normalizer 3.3.2
comm               0.2.2
contourpy          1.2.1
cycler             0.12.1
debugpy            1.8.1
decorator          5.1.1
deepdish           0.3.7
dill               0.3.8
docutils           0.21.2
executing          2.0.1
fonttools          4.53.0
fortranformat      2.0.0
future             1.0.0
h5py               3.11.0
idna               3.7
importlib_metadata 7.1.0
ipykernel          6.29.4
ipython            8.24.0
jaraco.classes     3.4.0
jaraco.context     5.3.0
jaraco.functools   4.0.1
jedi               0.19.1
jupyter_client     8.6.1
jupyter_core       5.7.2
keyring            25.2.1
kiwisolver         1.4.5
libcst             1.3.1
lmfit              1.3.1
madrigalWeb        3.3
markdown-it-py     3.0.0
matplotlib         3.9.0
matplotlib-inline  0.1.7
mdurl              0.1.2
more-itertools     10.2.0
msgpack            1.0.8
ndindex            1.8
nest-asyncio       1.6.0
netCDF4            1.7.1
nh3                0.2.17
numexpr            2.10.1
numpy              2.0.0
packaging          24.1
pandas             2.2.2
parso              0.8.4
pathlib2           2.3.7.post1
pexpect            4.9.0
pillow             10.3.0
pip                24.0
pkginfo            1.11.0
platformdirs       4.2.1
prompt-toolkit     3.0.43
psutil             5.9.8
ptyprocess         0.7.0
pure-eval          0.2.2
py-cpuinfo         9.0.0
pydarn             4.0
pydarnio           1.2.1
Pygments           2.18.0
pyIGRF             0.3.3
PyIRI              0.0.2
PyIRTAM            0.0.4
pyparsing          3.1.2
pyproj             3.6.1
pyproject_hooks    1.1.0
PyRAY              0.0.1
pyshp              2.3.1
python-dateutil    2.9.0.post0
pytz               2024.1
PyYAML             6.0.1
pyzmq              26.0.3
readme_renderer    43.0
requests           2.32.3
requests-toolbelt  1.0.0
rfc3986            2.0.0
rich               13.7.1
scipy              1.13.1
shapely            2.0.4
six                1.16.0
stack-data         0.6.3
tables             3.9.2
tornado            6.4
tqdm               4.66.2
traitlets          5.14.3
twine              5.1.0
tzdata             2024.1
uncertainties      3.1.7
urllib3            2.2.1
wcwidth            0.2.13
wheel              0.43.0
xcrun              0.4
zipp               3.19.2

**Comment by RemingtonRohel** [comment]
I was able to reproduce your error. It should be fixed by installing the `pydarnio` dependency from its develop branch, which can be done by 

```pip install git+https://github.com/SuperDARN/pydarnio.git@develop```

I would also recommend you make a virtual environment for installing software when working on a project, so that your dependencies won't be impacted by installing dependencies for other projects. There are some steps to follow [here](https://pydarn.readthedocs.io/en/main/user/install/#virtual-environments) that will guide you through it. If you follow these steps, you will have to also re-install pydarnio from the develop branch within that virtual environment.

Let me know if this fixes your problem.

**Comment by billetd** [comment]
I just posted an issue about this on pydarnio: https://github.com/SuperDARN/pyDARNio/issues/71

It should be fixed once develop is merged into main there.

**Comment by victoriyaforsythe** [comment]
Yay! That helped. Thank you so much for looking into it!

**Comment by carleyjmartin** [comment]
Thanks everyone, closing this as it's a pydarnio issue now.
